packages/sanity/src/structure/panes/document/documentPanel/documentViews/FormView.tsx

```diff
@@ -56,7 +56,8 @@ export const FormView = forwardRef<HTMLDivElement, FormViewProps>(function FormV
   } = useDocumentPane()
   const documentStore = useDocumentStore()
   const presence = useDocumentPresence(documentId)
-  const {title} = useDocumentTitle()
+  const currentTitle = useDocumentTitle()
+  const {title} = value ?? currentTitle
```

Comment:
I followed this code, as I was wondering why `title as string` is necessary on line 191.

It looks like it's because we're now primarily grabbing `title` field from the document's value, rather than the `useDocumentTitle` hook. Doesn't this logic assume the document has a `title` field, which might not be the case?

Reply:
That is true, that is why I set the currentTitle as default. The issue with the way that has been is that it doesn't update the title when the revision changes if it doesn't use the value.

Though maybe the solution should be in a different place (like in the useDocumentTitle)...
I'll make a note of this and what I'll probably do is update the test to not rely on this title and instead rely on the input value

Reply:
https://linear.app/sanity/issue/SAPP-1958/issue-where-the-document-title-is-not-being-updated

Created the issue, reverting this and fixing the test